src/pybind/mgr/rook/module.py

```diff
@@ -232,16 +233,8 @@ def get_inventory(self, host_filter=None, refresh=False):
         for host_name, host_devs in devs.items():
             devs = []
             for d in host_devs:
-                dev = inventory.Device(
-                    path='/dev/' + d['name'],
-                    sys_api=dict(
-                        rotational='1' if d['rotational'] else '0',
-                        size=d['size']
-                    ),
-                    available=d['empty'],
-                    rejected_reasons=[] if d['empty'] else ['not empty'],
-                )
-                devs.append(dev)
+                if 'cephVolumeData' in d and d['cephVolumeData']:
+                    devs.append(inventory.Device.from_json(json.loads(d['cephVolumeData'])))
```

Comment:
```python
if 'cephVolumeData' in d and d['cephVolumeData']:
    ...
else:
    devs.append(inventory.Device(
        path='/dev/' + d['name'],
        sys_api=dict(	
            rotational='1' if d['rotational'] else '0',	
            size=d['size']	
            ),
    )
```
? just as a fallback?

Reply:
Good idea. The only thing is that we are "masking" a problem getting the information from "ceph-volume" which must be the only source of truth. 
Because of that, I have explicitly set "available= False" with the explanation "device data coming from ceph-volume not provided".